src/interactions/panZoomInteraction.ts

```diff
@@ -20,56 +29,150 @@ export module Interaction {
      */
     constructor(xScale?: Scale.AbstractQuantitative<any>, yScale?: Scale.AbstractQuantitative<any>) {
       super();
-      if (xScale) {
-        this._xScale = xScale;
-        // HACKHACK #1388: self-register for resetZoom()
-        this._xScale.broadcaster.registerListener("pziX" + this.getID(), () => this.resetZoom());
-      }
-      if (yScale) {
-        this._yScale = yScale;
-        // HACKHACK #1388: self-register for resetZoom()
-        this._yScale.broadcaster.registerListener("pziY" + this.getID(), () => this.resetZoom());
-      }
-    }
+      this._xScale = xScale;
+      this._yScale = yScale;
 
-    /**
-     * Sets the scales back to their original domains.
-     */
-    public resetZoom() {
-      // HACKHACK #254
-      this._zoom = d3.behavior.zoom();
-      if (this._xScale) {
-        this._zoom.x((<any> this._xScale)._d3Scale);
-      }
-      if (this._yScale) {
-        this._zoom.y((<any> this._yScale)._d3Scale);
-      }
-      this._zoom.on("zoom", () => this._rerenderZoomed());
-      this._zoom(this._hitBox);
+      this._dragInteraction = new Interaction.Drag();
+      this._setupDragInteraction();
+      this._touchIds = d3.map();
     }
 
     public _anchor(component: Component.AbstractComponent, hitBox: D3.Selection) {
       super._anchor(component, hitBox);
-      this.resetZoom();
+      this._dragInteraction._anchor(component, hitBox);
+
+      var mouseDispatcher = Dispatcher.Mouse.getDispatcher(<SVGElement> this._componentToListenTo.content().node());
+      mouseDispatcher.onWheel("Interaction.PanZoom" + this.getID(), (p: Point, e: WheelEvent) => this._handleWheelEvent(p, e));
+
+      this._touchDispatcher = Dispatcher.Touch.getDispatcher(<SVGElement> this._componentToListenTo.content().node());
+      this._touchDispatcher.onTouchStart("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handlePinchStart(ids, idToPoint, e));
+      this._touchDispatcher.onTouchMove("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handlePinch(ids, idToPoint, e));
+      this._touchDispatcher.onTouchEnd("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handlePinchFinish(ids, idToPoint, e));
+      this._touchDispatcher.onTouchCancel("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handlePinchFinish(ids, idToPoint, e));
     }
 
-    public _requiresHitbox() {
-      return true;
+    private _handlePinchStart(ids: number[], idToPoint: { [id: number]: Point; }, e: TouchEvent) {
```

Comment:
Also since it doesn't actually process a pinch gesture, `_handleTouchStart` is probably the correct name for this method. Sorry to flip-flop on this one...